src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,93 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private static final double COEFFICIENT = 0.5;
+    private Object[] array = new Object[CAPACITY];
+    private int size = 0;
 
     @Override
     public void add(T value) {
-
+        if (size < array.length) {
+            array[size] = value;
+            size++;
+            return;
+        }
+        Object[] oldArray = new Object[array.length];
+        System.arraycopy(array, 0, oldArray, 0, array.length);
+        array = new Object[(int) (array.length * COEFFICIENT) + array.length];
+        System.arraycopy(oldArray, 0, array, 0, oldArray.length);
```

Comment:
try creating the new larger array and use one System.arraycopy()
then make `array` field point on the new array